src/ngx_http_lua_worker_thread.c

```diff
@@ -0,0 +1,437 @@
+/*
+ * Copyright (C) Yichun Zhang (agentzh)
+ * Copyright (C) Jinhua Luo (kingluo)
+ * I hereby assign copyright in this code to the lua-nginx-module project,
+ * to be licensed under the same terms as the rest of the code.
+ */
+
+
+#ifndef DDEBUG
+#define DDEBUG 0
+#endif
+#include "ddebug.h"
+
+
+#include "ngx_http_lua_worker_thread.h"
+#include "ngx_http_lua_util.h"
+
+
+#if (NGX_THREADS)
+
+
+#include <ngx_thread.h>
+#include <ngx_thread_pool.h>
+
+
+typedef struct ngx_http_lua_task_ctx_s {
+    lua_State                        *vm;
+    ngx_thread_mutex_t                mutex;
+    struct ngx_http_lua_task_ctx_s   *next;
+} ngx_http_lua_task_ctx_t;
+
+
+typedef struct {
+    ngx_http_lua_task_ctx_t *ctx;
+    ngx_http_lua_co_ctx_t   *wait_co_ctx;
+    ngx_http_cleanup_pt      cleanup;
+    void                    *cleanup_data;
+    int                      n_args;
+    int                      rc;
+    int                      is_abort;
+} ngx_http_lua_my_thread_ctx_t;
+
+
+static  ngx_http_lua_task_ctx_t   dummy_ctx;
+static  ngx_http_lua_task_ctx_t  *ctxpool = &dummy_ctx;
+
+
+static ngx_http_lua_task_ctx_t *
+ngx_http_lua_get_task_ctx(lua_State *L)
+{
+    ngx_http_lua_task_ctx_t *ctx = NULL;
+
+    size_t           path_len;
+    const char      *path;
+    size_t           cpath_len;
+    const char      *cpath;
+    lua_State       *vm;
+
+    if (ctxpool->next == NULL) {
```

Comment:
so, we can create many many Lua vms without any limitation?
if yes, that's sad.

Reply:
No, it's limited by the thread number of `thread_pool`.